bundles/org.openhab.binding.icalendar/src/main/java/org/openhab/binding/icalendar/internal/handler/EventFilterHandler.java

```diff
@@ -0,0 +1,362 @@
+            if (currentGroupId == null) {
+                return true;
+            }
+            for (ResultChannelSet channelSet : resultChannels) {
+                if (channelSet.resultGroup.getId().contentEquals(currentGroupId)) {
+                    return false;
+                }
+            }
+            return true;
+        }).forEach((Channel toDelete) -> {
+            thingBuilder.withoutChannel(toDelete.getUID());
+        });
+
+        resultChannels.stream().filter((ResultChannelSet current) -> {
+            return (getThing().getChannelsOfGroup(current.resultGroup.toString()).size() == 0);
+        }).forEach((ResultChannelSet current) -> {
+            for (ChannelBuilder builder : handlerCallback.createChannelBuilders(current.resultGroup, GROUP_TYPE_UID)) {
+                Channel currentChannel = builder.build();
+                @Nullable
+                Channel existingChannel = getThing().getChannel(currentChannel.getUID());
+                if (existingChannel == null) {
+                    thingBuilder.withChannel(currentChannel);
+                }
+            }
+        });
+
+        updateThing(thingBuilder.build());
+    }
+
+    /**
+     * Updates all states and channels.
+     */
+    private void updateStates() {
+        final Bridge iCalendarBridge = getBridge();
+        if (iCalendarBridge == null) {
+            logger.debug("Bridge not instantiated!");
+            return;
+        }
+        final ICalendarHandler iCalendarHandler = (ICalendarHandler) iCalendarBridge.getHandler();
+        if (iCalendarHandler == null) {
+            logger.debug("ICalendarHandler not instantiated!");
+            return;
+        }
+        final EventFilterConfiguration config = configuration;
+        if (config == null) {
+            logger.debug("Configuration not instantiated!");
+            return;
+        }
+        @Nullable
+        final AbstractPresentableCalendar cal = iCalendarHandler.getRuntimeCalendar();
+        if (cal != null) {
+            updateStatus(ThingStatus.ONLINE);
+
+            String textFilterValue = config.textEventValue;
+            EventTextFilter filter = null;
+            if (textFilterValue != null) {
+                try {
+                    if (config.textEventField == null || config.textValueType == null) {
+                        throw new IllegalArgumentException("Config is broken.");
+                    }
+                    EventTextFilter.Field textFilterField = EventTextFilter.Field.valueOf(config.textEventField);
+                    EventTextFilter.Type textFilterType = EventTextFilter.Type.valueOf(config.textValueType);
+
+                    filter = new EventTextFilter(textFilterField, textFilterValue, textFilterType);
+                } catch (IllegalArgumentException e) {
+                    logger.warn("Text-filter settings are not set properly.");
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR);
```

Comment:
See above. Remove logging and append the message to `updateStatus()`.

Reply:
OK